### PR TITLE
[GeneratorBundle] Add Groundcontrol-simple

### PR DESCRIPTION
**Pull request by dbeerten**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |

### CODE REVIEW

File: src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php [hunk 1]
```diff
@@ -30,70 +30,85 @@ class LayoutGenerator extends KunstmaanGenerator
...
-     * Generate the bower configuration files.
+     * Generate the groundcontrol(gulp) configuration files.
      */
-    private function generateBowerFiles()
+    private function generateGroundcontrolFiles()
     {
         $this->renderFiles(
-            $this->skeletonDir . '/bower/',
+            $this->skeletonDir . '/groundcontrol/bin/',
+            $this->rootDir . '/bin/frontend/',
```
Comment (dbeerten) [inline src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php]: This was a proposal from Ben and Stijn, but I don't know if this is the ideal location for de groundcontrol files..
@Devolicious @diskwriter  Any thoughts?

Comment (Devolicious) [inline src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php]: IMHO, it fits better in a `build` folder. The `bin` folder is for executables.

Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php]: I agree with @Devolicious , `bin` is for executables

File: docs/03-00-installation.md [hunk 2]
```diff
@@ -129,9 +126,9 @@ Then execute the following commands:
 
 ```
 bundle install
+nvm install
```
Comment (diskwriter) [inline docs/03-00-installation.md]: This requires you to have nvm previously installed, we might not want to enforce this.

File: src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php [hunk 3]
```diff
@@ -30,70 +30,92 @@ class LayoutGenerator extends KunstmaanGenerator
...
+            '.eslintrc',
             array('bundle' => $this->bundle, 'demosite' => $this->demosite),
             true
         );
         $this->renderSingleFile(
-            $this->skeletonDir . '/bower/',
+            $this->skeletonDir . '/groundcontrol/',
             $this->rootDir,
-            '.bowerrc',
+            '.nvmrc',
```
Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Generator/LayoutGenerator.php]: See previous comment on nvm

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/CookieConsent.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+export default class CookieConsent {
+
+    constructor() {
+        this.cookieBar = document.getElementById('cookie-bar');
+        this.cookieBarConsentBtn = document.getElementById('cookie-bar__consent-btn');
+        this.hasCookie = document.cookie.match(/(?:(?:^|.*;\s*)bundles-cookie-consent\s*=\s*([^;]*).*$)|^.*$/)[1];
```
Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/CookieConsent.js]: These can just be `const` I think, there's no need for them to be public.

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
+export default class Videolink {
+
+    constructor() {
+        Array.from(document.querySelectorAll('.js-videolink-placeholder')).forEach((placeholder) => {
```
Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js]: `Array.from` is not supported in IE and is one of those Array prototype methods that Babel doesn't catch if you use the es2015 preset. A better solution would be to use `Array.prototype.slice.call(...)`. 
(Or look for a preset that transpiles Array prototype methods)

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            const provider = placeholder.parentNode.getAttribute('data-video-provider');
```
Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js]: Maybe do a `hasAttribute` check first

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                requestUrl = `http://vimeo.com/api/v2/video/${id}.json`;
+            } else if (provider === 'dailymotion') {
+                requestUrl = `https://api.dailymotion.com/video/${id}?fields=thumbnail_720_url`;
+            }
+
+            request.open('GET', requestUrl, true);
+
+            request.onload = () => {
+                if (request.status >= 200 && request.status < 400) {
+                    json = JSON.parse(request.responseText);
```
Comment (diskwriter) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/layout/Resources/ui/js/Videolink.js]: Nitpicking: It's odd that a let named json doesn't contain json but an object

File: docs/03-01-system-requirements.md [hunk 8]
```diff
@@ -26,12 +26,9 @@
         * session.auto_start = Off
         * date.timezone should be configured
 * MySQL 5.x (some work on sqlite has been done but is not supported)
-* Node.js and NPM with these packages installed globally:
-    * bower
-    * gulp
+* Node.js (7.7) and NPM with these packages installed globally:
     * uglify-js
     * uglifycss
-* Ruby, RubyGems and the bundler gem
```
Comment (diskwriter) [inline docs/03-01-system-requirements.md]: Aren't these still required for hologram?

Comment (dbeerten) [inline docs/03-01-system-requirements.md]: Only if you want to build the hologram style guide of the frontend styling, so not really required for the bundles I guess

**Pull request merged**